src/main/java/mods/railcraft/common/blocks/tracks/TrackNextGenLocking.java

```diff
@@ -324,12 +390,25 @@ public void readFromNBT(NBTTagCompound data) {
             profile = LockingProfileType.fromOrdinal(data.getByte("profile"));
         profileInstance = profile.create(this);
         profileInstance.readFromNBT(data);
-        powered = data.getBoolean("powered");
-        delay = data.getInteger("delay");
-        reset = data.getByte("reset");
-
-        if (data.hasKey("uuidHigh"))
-            uuid = new UUID(data.getLong("uuidHigh"), data.getLong("uuidLow"));
+        redstone = data.getBoolean("powered");
+        
+        if(data.hasKey("locked"))
+        	locked = data.getBoolean("locked");
+                
+        if(data.hasKey("trainLeaving"))
+        	trainLeaving = data.getBoolean("trainLeaving");
+                
+        if(data.hasKey("trainDelay"))
+        	trainDelay = data.getInteger("trainDelay");        
+
+        prevCartUUID = readUUID("prevCart", data);  
+        currentCartUUID = readUUID("currentCart", data);
+        
+        uuid = readUUID("uuid", data);
+        
+        justLoaded = true; // This signals updateEntity() to dereference the cart UUID's we read in here
+        
+        System.out.println("Read from NBT");
```

Comment:
*cough* You might want to remove this ;)

FYI, you don't need to create a new PR, just make a commit to your branch.